BlueDBOnDisk/src/main/java/org/bluedb/disk/file/ReadWriteFileManager.java

```diff
@@ -103,14 +103,18 @@ public void makeCopy(Path srcPath, Path destPath) throws BlueDbException, IOExce
 				destMetadata.put(BlueFileMetadataKey.ENCRYPTION_VERSION_KEY, encryptionService.getCurrentEncryptionVersionKey());
 			}
 			writeMetadata(destMetadata, dos, destPath);
-			
+
 			// Write bytes
 			byte[] rawBytes = FileUtils.readAllBytes(dis);
 			if (EncryptionUtils.shouldWriterSkipEncryptionForUnchangedDataUsingRawBytes(srcMetadata, destMetadata)) {
 				dos.write(rawBytes);
 			} else {
-				byte[] unencryptedBytes = encryptionService.decryptOrReturn(srcMetadata, rawBytes);
-				dos.write(unencryptedBytes);
+				byte[] bytes = encryptionService.decryptOrReturn(srcMetadata, rawBytes);
+				if (destMetadata.containsKey(BlueFileMetadataKey.ENCRYPTION_VERSION_KEY)) {
+					String encryptionVersionKey = destMetadata.get(BlueFileMetadataKey.ENCRYPTION_VERSION_KEY);
+					bytes = encryptionService.encryptOrThrow(encryptionVersionKey, bytes);
+				}
+				dos.write(bytes);

```

Comment:
This bug wasn't being found by your automated tests, so I'd prefer to add or modify a test to cover that case that was missed along with fixing the problem.

Reply:
Good call, I'll do that. The integration tests should hopefully catch major bugs like this as well, but I'll update a unit test or write a new one to be sure we catch this error in the primary test suite.